### ISSUE TITLE
SerializerError with unescaped quotes on raw fields

### ISSUE THREAD

**Issue by jminuscula** [elastic/elasticsearch-py#576]
Hi,

we're having trouble trying to fetch values which contain double quotes in a not analyzed index. Given our mapping and this sample document

```
"mappings": {
    "contact": {
        "dynamic_templates": [
            {"raw_field": {
                "path_match": "fields.*",
                "match_mapping_type": "string",
                "mapping": {
                    "type": "string",
                    "index": "analyzed",
                    "fields": {
                        "raw": {
                            "type": "string",
                            "index": "not_analyzed"
                        }
                    }
                }
            }},
        ]
    },
    ...
}
```

```
{
    'fields': {
        'my_field': 'my name is "John"'
     }
}
```

The error looks something like this:

```
Traceback (most recent call last):
      File "/home/ubuntu/analytics/.venv/lib/python3.5/site-packages/elasticsearch/serializer.py", line 38, in loads
        return json.loads(s)
      File "/usr/lib/python3.5/json/__init__.py", line 319, in loads
        return _default_decoder.decode(s)
      File "/usr/lib/python3.5/json/decoder.py", line 339, in decode
        obj, end = self.raw_decode(s, idx=_w(s, 0).end())
      File "/usr/lib/python3.5/json/decoder.py", line 355, in raw_decode
        obj, end = self.scan_once(s, idx)
    json.decoder.JSONDecodeError: Expecting ',' delimiter: line 1 column 8699 (char 8698)

During handling of the above exception, another exception occurred:
    
    Traceback (most recent call last):
      File "/home/ubuntu/analytics/app/lib/serializers.py", line 40, in loads
        return super(EscapingJSONSerializer, self).loads(s)
      File "/home/ubuntu/analytics/.venv/lib/python3.5/site-packages/elasticsearch/serializer.py", line 40, in loads
        raise SerializationError(s, e)
    elasticsearch.exceptions.SerializationError:

[...] "buckets":[{"key":"my name is "John"","doc_count":1}]}}}', JSONDecodeError("Expecting ',' delimiter: line 1 column 8699 (char 8698)",))
    
    During handling of the above exception, another exception occurred:
    
    Traceback (most recent call last):
      File "/home/ubuntu/analytics/.venv/lib/python3.5/site-packages/elasticsearch/serializer.py", line 38, in loads
        return json.loads(s)
      File "/usr/lib/python3.5/json/__init__.py", line 319, in loads
        return _default_decoder.decode(s)
      File "/usr/lib/python3.5/json/decoder.py", line 339, in decode
        obj, end = self.raw_decode(s, idx=_w(s, 0).end())
      File "/usr/lib/python3.5/json/decoder.py", line 355, in raw_decode
        obj, end = self.scan_once(s, idx)
    json.decoder.JSONDecodeError: Expecting property name enclosed in double quotes: line 1 column 8710 (char 8709)

[...]
```

another series of serialization errors follow. 

We're not entirely sure how we should be dealing with this, since indexing the python dictionary works without a problem. Can we make ES escape the values prior to JSON serialization? Is there something we're missing? Thanks!

We're using `elasticsearch-py 5.0.1` and `ES v2.3.2`.

**Comment by HonzaKral** [comment]
Hi,
when does the error occur, during which operation? Could you please share the code? I am unable to replicate this issue.

Also please note that the combination of versions you are using is not supported - please use a `2.x` version of the library with `2.x` version of elasticsearch.

Thanks!

**Comment by jminuscula** [comment]
Hi Honza,

this is happening at the `transport` right after executing a query. Here:
https://github.com/elastic/elasticsearch-py/blob/5.0/elasticsearch/transport.py#L357

After much debugging, I think this is caused by the `filter_path` parameter. If I remove it from the query, the response is escaped properly and the serialization does not fail.

This is what my trace looks like:

```
ipdb> type(self.current_search)
<class 'elasticsearch_dsl.search.Search'>

ipdb> self.current_search.to_dict()
{'query': {'match_all': {}}, 'aggs':
    {'by_interval': {'aggs': {'id_cardinality': {'cardinality': {'field': 'id'}}}, 'date_histogram': {'time_zone': 'UTC', 'extended_bounds': {}, 'interval': 'month', 'field': 'created_on', 'min_doc_count': 0}}
     'contact_field_gender': {'terms': {'size': 5, 'field': 'fields.gender.raw'}},
     'contact_field_age': {'terms': {'size': 5, 'field': 'fields.age.raw'}},
     'contact_field_locale_language': {'terms': {'size': 5, 'field': 'fields.locale_language.raw'}}, 
     'contact_field_locale_country': {'terms': {'size': 5, 'field': 'fields.locale_country.raw'}}, 
     'contact_field_last_referral': {'terms': {'size': 5, 'field': 'fields.last_referral.raw'}}, 
     'contact_field_first_referral': {'terms': {'size': 5, 'field': 'fields.first_referral.raw'}},
     'contact_field_gender_unknown': {'filter': {'bool': {'should': [{'bool': {'must_not': [{'exists': {'field': 'fields.gender.raw'}}]}}, {'term': {'fields.gender.raw': ''}}]}}},
     'contact_field_locale_country_unknown': {'filter': {'bool': {'should': [{'bool': {'must_not': [{'exists': {'field': 'fields.locale_country.raw'}}]}}, {'term': {'fields.locale_country.raw': ''}}]}}}, 
     'contact_field_first_referral_unknown': {'filter': {'bool': {'should': [{'bool': {'must_not': [{'exists': {'field': 'fields.first_referral.raw'}}]}}, {'term': {'fields.first_referral.raw': ''}}]}}},
     'contact_field_age_unknown': {'filter': {'bool': {'should': [{'bool': {'must_not': [{'exists': {'field': 'fields.age.raw'}}]}}, {'term': {'fields.age.raw': ''}}]}}},
     'contact_field_locale_language_unknown': {'filter': {'bool': {'should': [{'bool': {'must_not': [{'exists': {'field': 'fields.locale_language.raw'}}]}}, {'term': {'fields.locale_language.raw': ''}}]}}}, 
     'contact_field_last_referral_unknown': {'filter': {'bool': {'should': [{'bool': {'must_not': [{'exists': {'field': 'fields.last_referral.raw'}}]}}, {'term': {'fields.last_referral.raw': ''}}]}}}
    }
}

ipdb> self.current_search._params
{'filter_path': ['hits.total', 'aggregations.by_interval.*.doc_count', 'aggregations.by_interval.*.key', 'aggregations.by_interval.*.id_cardinality', 'aggregations.contact_field*.*']}


ipdb> self.current_search.execute()
```

and this is the full traceback

```
Traceback (most recent call last):
  File "/Users/jacobo/reply/analytics/.venv/lib/python3.4/site-packages/elasticsearch/serializer.py", line 38, in loads
    return json.loads(s)
  File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/json/__init__.py", line 318, in loads
    return _default_decoder.decode(s)
  File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/json/decoder.py", line 343, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/json/decoder.py", line 359, in raw_decode
    obj, end = self.scan_once(s, idx)
ValueError: Expecting ',' delimiter: line 1 column 154 (char 153)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/jacobo/reply/analytics/app/lib/queue.py", line 81, in resolve
    action_future.set_result(future.result())
  File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/concurrent/futures/_base.py", line 395, in result
    return self.__get_result()
  File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/concurrent/futures/_base.py", line 354, in __get_result
    raise self._exception
  File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/concurrent/futures/thread.py", line 54, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/Users/jacobo/reply/analytics/app/lib/queue.py", line 74, in run_action
    response = self.get_stats_formatted(**filters)
  File "/Users/jacobo/reply/analytics/app/lib/queue.py", line 183, in get_stats_formatted
    ('contacts', self.get_contact_stats(index, org, interval, date_range)),
  File "/Users/jacobo/reply/analytics/app/lib/queue.py", line 143, in get_contact_stats
    return contacts.execute_query()
  File "/Users/jacobo/reply/analytics/app/lib/search.py", line 363, in execute_query
    return self.current_search.execute()
  File "/Users/jacobo/reply/analytics/.venv/lib/python3.4/site-packages/elasticsearch_dsl/search.py", line 627, in execute
    **self._params
  File "/Users/jacobo/reply/analytics/.venv/lib/python3.4/site-packages/elasticsearch/client/utils.py", line 69, in _wrapped
    return func(*args, params=params, **kwargs)
  File "/Users/jacobo/reply/analytics/.venv/lib/python3.4/site-packages/elasticsearch/client/__init__.py", line 539, in search
    doc_type, '_search'), params=params, body=body)
  File "/Users/jacobo/reply/analytics/.venv/lib/python3.4/site-packages/elasticsearch/transport.py", line 357, in perform_request
    data = self.deserializer.loads(data, headers.get('content-type'))
  File "/Users/jacobo/reply/analytics/.venv/lib/python3.4/site-packages/elasticsearch/serializer.py", line 76, in loads
    return deserializer.loads(s)
  File "/Users/jacobo/reply/analytics/.venv/lib/python3.4/site-packages/elasticsearch/serializer.py", line 40, in loads
    raise SerializationError(s, e)
elasticsearch.exceptions.SerializationError ...
```

sorry for the confusion —this is happening on our client, which uses 
```
elasticsearch==2.4.0
elasticsearch-dsl==2.1.0
```

Thanks!

**Comment by HonzaKral** [comment]
Thank you for the information, this is a known issue that has been fixed in jackson and resolved in elasticsearch `5.0.0` - https://github.com/elastic/elasticsearch/issues/18076
